naga/src/back/spv/block.rs

```diff
@@ -1563,6 +1564,45 @@ impl BlockContext<'_> {
                                 ));
                                 extracted = casted;
                             }
+                            if should_clamp {
+                                let (min, min_op, max, max_op) = if is_signed {
+                                    (
+                                        crate::Literal::I32(-128),
+                                        spirv::GLOp::SMin,
+                                        crate::Literal::I32(127),
+                                        spirv::GLOp::SMax,
```

Comment:
We could use `SClamp`/`UClamp`.

Reply:
Ah, yeah. I did it this way to imitate other code I saw for clamping; is there maybe a question of availability for this op? Otherwise, I'd be happy to refactor the other site in a follow-up. 😀

Reply:
Ah, I just reloaded this from our implementation of `MathFunction::Clamp`:

```rust
                        // Clamp is undefined if min > max. In practice this means it can use a median-of-three
                        // instruction to determine the value. This is fine according to the WGSL spec for float
                        // clamp, but integer clamp _must_ use min-max. As such we write out min/max.

```

…but that's fine, because we control the min. and max. parameters for clamped packing specifically. Woot! Fixed up with b2911321.

Reply:
D'oh, and added the updated snapshot result with b09dd786. 😅